src/components/Recipe.js

```diff
@@ -1,15 +1,40 @@
 import { useState } from 'react';
-import { Card, Modal } from 'react-bootstrap';
+import { Card, Modal, Form } from 'react-bootstrap';
 import '../styles/Recipe.css';
 
 export default function Recipe(props) {
+  const { setCheckedRecipe, data } = props;
   const [modal, setModal] = useState(false);
-  const { method, ingredients, name, createdBy, time } = props.data;
+  const { _id, method, ingredients, name, createdBy, time } = data;
+
+  const onCheckbox = (target) => {
+    setCheckedRecipe((current) => {
```

Comment:
I'm not sure i understand this. This is the component for a single recipe? and it's tracking whether it's checked or not? what does it do if it's checked? It makes more sense to me for this to go inside RecipeView instead

Reply:
The idea is this: Users select a recipe by clicking the checkbox near the top right corner of the recipe. Upon doing so, they have the option to edit or delete that single recipe. The other code within this function is just to automatically unselect previously selected recipe when user selects a different recipe. Of course, if user unselects currently selected recipe, then the options to edit or delete disappears. Below are images demonstrating it:

1. When you first load the my recipes page from the dropdown menu
![image](https://github.com/Lord-Bubbles/recipe-app/assets/24702602/05273a31-b89a-4617-8efb-7f0916269137)
2. When you select a recipe to edit or delete.
![image](https://github.com/Lord-Bubbles/recipe-app/assets/24702602/4a312569-d830-4861-9c62-a9786da8e99a)
3. When you select a different recipe without unselecting recipe from process 2.
![image](https://github.com/Lord-Bubbles/recipe-app/assets/24702602/543138d4-db15-48b0-b625-bb433dc6b4f1)
4. When you unselect newly selected recipe
![image](https://github.com/Lord-Bubbles/recipe-app/assets/24702602/dd41e85b-4a98-4cce-95d1-53f49dfa54c5)

Thus, it made sense in my head to put these in each recipe component and the myrecipes page remembers the currently selected recipe and decides whether to display the options or not. 

Should I keep with this checkbox idea or do something like a toggeable button to select a recipe?

Reply:
I think we should move this logic (the part where only one item is checked) into MyRecipes.js, where you're keeping track of whether to display edit or not and which recipe to edit.

I also currently don't know how or why this works. you're calling setCheckedRecipe with a function (current) => {...}, but setCheckedRecipe wants a recipe? so recipe is of type function? What's calling the function to set current?

Reply:
What I need is for every recipe to be able to be checked and saved in the checkedRecipe state. This is why the logic of setting the checkedRecipe state is in the Recipe component, not the MyRecipes component. Within the Recipe component, I call the setCheckedRecipe function to determine the new state of checkedRecipe.  To gain access to this function in Recipe component, within the MyRecipes component, I pass the setCheckedRecipe as a function prop all the way down to the Recipe component. This way when the checkbox changes within the Recipe component, the state stored in MyRecipes also updates, triggering the display you see in the images above. 

Within react, there is a function form of setStateName where you have access to the current state and you can return a new state based off of it. This is the form that I'm using. If the current state is null, then I set the state to an object containing the checkbox element and the id of the recipe. The recipe id will be used to easily retrieve and update recipe object during editing. Otherwise, i uncheck the checkbox from current state and return a similar object to the one described above. If the checkbox and newly checked checkbox is the same, then I set the state to null (I no longer have anything selected). 

If there is anything that's still unclear, please let me know.

Reply:
i think my point is missed here but it's not important enough for me to push this